operator/controller/src/test/java/io/apicurio/registry/operator/it/EnvITTest.java

```diff
@@ -0,0 +1,177 @@
+                    appEnv = c.getEnv();
+                }
+            }
+            assertThat(appEnv).map(EnvVar::getName).containsOnlyOnce(defaultAppEnv);
+            var QUARKUS_HTTP_ACCESS_LOG_ENABLED = appEnv.stream().filter(e -> "QUARKUS_HTTP_ACCESS_LOG_ENABLED".equals(e.getName())).map(EnvVar::getValue).findAny().orElse(null);
+            assertThat(QUARKUS_HTTP_ACCESS_LOG_ENABLED).isEqualTo("true");
+
+            List<EnvVar> uiEnv = null;
+            var uiContainers = client.apps().deployments().inNamespace(namespace).withName(registry.getMetadata().getName() + "-ui-deployment").get().getSpec().getTemplate().getSpec().getContainers();
+            for (var c : uiContainers) {
+                if (UI_CONTAINER_NAME.equals(c.getName())) {
+                    uiEnv = c.getEnv();
+                }
+            }
+            assertThat(uiEnv).map(EnvVar::getName).containsOnlyOnce(defaultUIEnv);
+            var REGISTRY_API_URL = uiEnv.stream().filter(e -> "REGISTRY_API_URL".equals(e.getName())).map(EnvVar::getValue).findAny().orElse(null);
+            assertThat(REGISTRY_API_URL).startsWith("http://");
+
+            return true;
+        });
+
+        registry.getSpec().getApp().setEnv(List.of(
+                new EnvVarBuilder().withName("APP_VAR_1_NAME").withValue("APP_VAR_1_VALUE").build(),
+                new EnvVarBuilder().withName("QUARKUS_HTTP_ACCESS_LOG_ENABLED").withValue("false").build(),
+                new EnvVarBuilder().withName("APP_VAR_2_NAME").withValue("APP_VAR_2_VALUE").build()
+        ));
+        registry.getSpec().getUi().setEnv(List.of(
+                new EnvVarBuilder().withName("UI_VAR_1_NAME").withValue("UI_VAR_1_VALUE").build(),
+                new EnvVarBuilder().withName("REGISTRY_API_URL").withValue("FOO").build(),
+                new EnvVarBuilder().withName("UI_VAR_2_NAME").withValue("UI_VAR_2_VALUE").build()
+        ));
+
+        client.resource(registry).update();
+
+        await().ignoreExceptions().until(() -> {
+
+            List<EnvVar> appEnv = null;
+            var appContainers = client.apps().deployments().inNamespace(namespace).withName(registry.getMetadata().getName() + "-app-deployment").get().getSpec().getTemplate().getSpec().getContainers();
+            for (var c : appContainers) {
+                if (APP_CONTAINER_NAME.equals(c.getName())) {
+                    appEnv = c.getEnv();
+                }
+            }
+            assertThat(appEnv).map(EnvVar::getName).containsOnlyOnce(defaultAppEnv);
+            var QUARKUS_HTTP_ACCESS_LOG_ENABLED = appEnv.stream().filter(e -> "QUARKUS_HTTP_ACCESS_LOG_ENABLED".equals(e.getName())).map(EnvVar::getValue).findAny().orElse(null);
+            assertThat(QUARKUS_HTTP_ACCESS_LOG_ENABLED).isEqualTo("false");
+            assertThat(appEnv).containsSubsequence(
+                    new EnvVarBuilder().withName("APP_VAR_1_NAME").withValue("APP_VAR_1_VALUE").build(),
```

Comment:
nitpick: maybe worth building those `EnvVar`s outside the test body to improve readability?